src/shogun/preprocessor/RandomFourierGaussPreproc.cpp

```diff
@@ -46,6 +45,7 @@ void RandomFourierGaussPreproc::fit(std::shared_ptr<Features> f)
 {
 	auto num_features = f->as<DenseFeatures<float64_t>>()->get_num_features();
 	require(num_features > 0, "Input space dimension must be greater than zero");
```

Comment:
Input space dimension (%d) must be greater than zero ... i.e. print the value so that users can debug

Reply:
Rather "Given input space dimension (%d) is less than zero"? Because "-3 must be greater than zero" doesn't sit well.

Reply:
I dont really agree. Imo it is better to state what is expected (greater than zero, as is there), and also provide what was given. Negations only confuse and that is an approach we usually take in shogun